File: javascript/7.js
const primes = function * () {
  const xs = []
  let x = 2
  while (xs.length < 10001) {
    if (xs.every(y => x % y !== 0)) {
      xs.push(x)
      yield x
    }
    x++
  }
}
console.log([...primes()].pop())

File: clojure/7.clj
(def prime-gen
  (let [primes (atom [])]
    (for [n (iterate inc 2)
            :when (not-any? #(zero? (mod n %))
                            (filter #(<= % (Math/sqrt n)) @primes))]
         (do (swap! primes conj n) n))))

(println (nth prime-gen 10000))
